io/net/sriov_device_test.py

```diff
@@ -0,0 +1,235 @@
+
+    @staticmethod
+    def get_partition_name(component):
+        '''
+        get partition name from lparstat -i
+        '''
+
+        for line in process.system_output('lparstat -i', ignore_status=True,
+                                          shell=True,
+                                          sudo=True).decode("utf-8") \
+                                                    .splitlines():
+            if component in line:
+                return line.split(':')[-1].strip()
+        return ''
+
+    def login(self, ipaddr, username, password):
+        '''
+        SSH Login method for remote server
+        '''
+        pxh = pxssh.pxssh(encoding='utf-8')
+        # Work-around for old pxssh not having options= parameter
+        pxh.SSH_OPTS = "%s  -o 'StrictHostKeyChecking=no'" % pxh.SSH_OPTS
+        pxh.SSH_OPTS = "%s  -o 'UserKnownHostsFile /dev/null' " % pxh.SSH_OPTS
+        pxh.force_password = True
+
+        pxh.login(ipaddr, username, password)
+        pxh.sendline()
+        pxh.prompt(timeout=60)
+        pxh.sendline('exec bash --norc --noprofile')
+        pxh.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        pxh.set_unique_prompt()
+        pxh.prompt(timeout=60)
+        self.pxssh = pxh
+
+    def run_command(self, command, timeout=300):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on peer lpar: %s", command)
+        if not hasattr(self, 'pxssh'):
+            self.fail("SSH Console setup is not yet done")
+        con = self.pxssh
+        con.sendline(command)
+        con.expect("\n")  # from us
+        con.expect(con.PROMPT, timeout=timeout)
+        output = "".join(con.before)
+        con.sendline("echo $?")
+        con.prompt(timeout)
+        exitcode = int(''.join(con.before.splitlines()[1:]))
+        if exitcode != 0:
+            raise CommandFailed(command, output, exitcode)
+        return output
+
+    def add_logical_sriov(self):
+        '''
+        create logical sriov device
+        '''
+        cmd = 'chhwres -r sriov -m %s --rsubtype logport \
+               -o a -p %s -a \"adapter_id=%s,phys_port_id=%s, \
+                logical_port_type=eth,mac_addr=%s\" ' \
+               % (self.server, self.lpar, self.adapter_id,
+                  self.sriov_port, self.mac_id)
+        try:
+            self.run_command(cmd)
+        except CommandFailed as cf:
+            self.fail("failed to create logical device" % cf)
+        cmd = "lshwres -r sriov --rsubtype logport -m  %s \
+               --level eth | grep %s | grep %s" \
+               % (self.server, self.lpar, self.mac_id)
+        output = self.run_command(cmd)
+        self.logical_port_id = output.split(',')[6].split('=')[-1]
+
+    def removed_logical_device(self):
```

Comment:
change `removed_logical_device` to `test_remove_sriov_device`
make it a test to help in scenario test

Reply:
ok, then in test add  i will add ping ..?